pkg/api/norman/store/service/service.go

```diff
@@ -33,12 +33,29 @@ func (p *Store) Create(apiContext *types.APIContext, schema *types.Schema, data
 			data["clusterIp"] = ""
 		}
 	}
+	// Check for unset ipFamilyPolicy, for headless services this defaults to dual stack but will not function if the cluster does not have dual stack properly configured (IPv6 CIDRs)
+	// If its not explicitly configured in the request, we default to SingleStack
 	if schema.ID == "service" {
-		logrus.Tracef("Service: Create: data.kind [%v], data.clusterIp [%v]", data["kind"], data["clusterIp"])
-		if data["kind"] == "ClusterIP" && data["clusterIp"] == nil {
-			if data["ipFamilyPolicy"] == nil {
-				logrus.Debugf("Setting ipFamilyPolicy to SingleStack for service name [%s] service kind [%s]", data["name"], data["kind"])
-				data["ipFamilyPolicy"] = "SingleStack"
+		logrus.Tracef("Service: Create: data [%v]", data)
+		if val, ok := data["kind"]; ok {
+			if val == "ClusterIP" {
+				var configureSingleStack bool
+				if val, ok := data["clusterIp"]; ok {
+					if val == nil || val == "None" {
+						if _, ok := data["ipFamilyPolicy"]; !ok {
```

Comment:
Nit, might be easier to combine
```
if val, ok := data["ipFamilyPolicy"]; !ok || val == nil {
  logrus.Debugf("Setting ipFamilyPolicy to SingleStack for service name [%s] service kind [%s]", data["name"], data["kind"])
  data["ipFamilyPolicy"] = "SingleStack"
}
```

Reply:
Nit: agree that it could be consolidated, but we also need to check the  `clusterIP`'s corresponding value